### PR TITLE
Update client generator to trim single quote in field names

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[3992](https://github.com/ballerina-platform/ballerina-standard-library/issues/3992)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 1]
```diff
@@ -1054,4 +1058,11 @@ private static String getEntityNameConstant(String entityName) {
         }
         return outputString.toString();
     }
+
+    private static String formatFieldName(String fieldName) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:1062]:
```suggestion
    private static String stripEscapeCharacter(String fieldName) {
```

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 2]
```diff
@@ -1054,4 +1058,11 @@ private static String getEntityNameConstant(String entityName) {
...
+        if (fieldName.startsWith("\'")) {
+            return fieldName.substring(1);
+        }
+        return fieldName;
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:1066]:
```suggestion
        return fieldName.startsWith("'") ? fieldName.substring(1) : fieldName;
```

**Review by daneshk — APPROVED** [review]

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java (hunk 2 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java]: updated in https://github.com/ballerina-platform/persist-tools/pull/117/commits/06860b7f7fa6a73ba0dfad04d67d653c21f63dd1

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java (hunk 1 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java]: updated in https://github.com/ballerina-platform/persist-tools/pull/117/commits/06860b7f7fa6a73ba0dfad04d67d653c21f63dd1

**Pull request merged**